### PR TITLE
optimeize unescape slash path

### PR DESCRIPTION
**Pull request by chenylh**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by mattklein123** [comment]
Seems fine but is this actually faster?

/wait

**Comment by nezdolik** [comment]
@yanavlasov friendly ping

File: source/common/http/path_utility.cc [hunk 1]
```diff
@@ -92,13 +82,14 @@ PathUtil::UnescapeSlashesResult PathUtil::unescapeSlashes(RequestHeaderMap& head
...
   const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
 
-  // TODO(yanavlasov): optimize this by adding case insensitive matcher
   std::string decoded_path{path};
-  unescapeInPath(decoded_path, "%2F", "/");
-  unescapeInPath(decoded_path, "%2f", "/");
-  unescapeInPath(decoded_path, "%5C", "\\");
-  unescapeInPath(decoded_path, "%5c", "\\");
-  headers.setPath(absl::StrCat(decoded_path, query));
+  std::map<const char*, const char*> replacements;
```
Comment (yanavlasov) [inline source/common/http/path_utility.cc:86]:
```suggestion
  static const std::vector<std::pair<absl::string_view, absl::string_view>> replacements{{"2F", "/"}, {"2f", "/"}, ...};
```

And then remove dynamic initialization below.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/common/http/path_utility.cc (hunk 1 above)
Comment (chenylh) [reply source/common/http/path_utility.cc:86]: At first, it was written like this, but the code format verification did not pass

Comment (chenylh) [reply source/common/http/path_utility.cc:86]: Ok

File: source/common/http/path_utility.cc [hunk 2]
```diff
@@ -92,13 +82,15 @@ PathUtil::UnescapeSlashesResult PathUtil::unescapeSlashes(RequestHeaderMap& head
...
-  unescapeInPath(decoded_path, "%5C", "\\");
-  unescapeInPath(decoded_path, "%5c", "\\");
-  headers.setPath(absl::StrCat(decoded_path, query));
+  static const std::vector<std::pair<absl::string_view, absl::string_view>> replacements{
+      {"%2F", "/"},
+      {"%2f", "/"},
+      {"%5C", "\\"},
+      {"%5c", "\\"},
+  };
+  headers.setPath(absl::StrCat(absl::StrReplaceAll(decoded_path, replacements), query));
```
Comment (yanavlasov) [inline source/common/http/path_utility.cc:92]: One more suggestion:

```suggestion
  headers.setPath(absl::StrCat(absl::StrReplaceAll(path, replacements), query));
```

And remove the `decoded_path` variable above.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
Thanks you so much. I just have one more small suggestion. Thanks.

/wait

File: source/common/http/path_utility.cc (hunk 2 above)
Comment (chenylh) [reply source/common/http/path_utility.cc:92]: OK

**Pull request merged**